src/diffusers/models/resnet.py

```diff
@@ -373,59 +408,89 @@ def __init__(
         self.in_channels = in_channels
         out_channels = in_channels if out_channels is None else out_channels
+        mid_channels = out_channels if mid_channels is None else mid_channels
         self.out_channels = out_channels
         self.use_conv_shortcut = conv_shortcut
-        self.time_embedding_norm = time_embedding_norm
         self.up = up
         self.down = down
         self.output_scale_factor = output_scale_factor
-
+        self.norm1_scale_shift = norm1_scale_shift
+        self.time_embedding_norm = time_embedding_norm
+        
         if groups_out is None:
             groups_out = groups
 
-        self.norm1 = torch.nn.GroupNorm(num_groups=groups, num_channels=in_channels, eps=eps, affine=True)
-
-        self.conv1 = torch.nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1, padding=1)
-
+        num_groups_in = groups if groups is not None else max(1, in_channels//group_size)
+        
+        if norm1_scale_shift and temb_channels is not None:
+                self.norm1 = AdaGroupNorm(temb_channels, in_channels, num_groups_in, eps=eps)
+        else:
+            self.norm1 = torch.nn.GroupNorm(num_groups=num_groups_in , num_channels=in_channels, eps=eps, affine=True)
+        
+        self.conv1 = torch.nn.Conv2d(in_channels, mid_channels, kernel_size=3, stride=1, padding=1)
+        
+        if dropout_after_conv:
+            self.dropout1 = torch.nn.Dropout(dropout, inplace=True)
+        else:
+            self.dropout1 = None
+       
+        num_groups_out = groups_out if groups_out is not None else max(1, mid_channels//group_size)
+        
         if temb_channels is not None:
-            if self.time_embedding_norm == "default":
-                time_emb_proj_out_channels = out_channels
+            if time_embedding_norm == "default":
+                self.time_emb_proj = torch.nn.Linear(temb_channels, mid_channels)
+                self.norm2 = torch.nn.GroupNorm(num_groups=num_groups_out, num_channels=mid_channels, eps=eps, affine=True)
             elif self.time_embedding_norm == "scale_shift":
-                time_emb_proj_out_channels = out_channels * 2
+                self.time_emb_proj = None
+                self.norm2 = AdaGroupNorm(temb_channels, mid_channels, num_groups_out, eps=eps)
             else:
                 raise ValueError(f"unknown time_embedding_norm : {self.time_embedding_norm} ")
-
-            self.time_emb_proj = torch.nn.Linear(temb_channels, time_emb_proj_out_channels)
         else:
             self.time_emb_proj = None
+            self.norm2 = torch.nn.GroupNorm(num_groups=num_groups_out, num_channels=mid_channels, eps=eps, affine=True)
 
-        self.norm2 = torch.nn.GroupNorm(num_groups=groups_out, num_channels=out_channels, eps=eps, affine=True)
-        self.dropout = torch.nn.Dropout(dropout)
-        self.conv2 = torch.nn.Conv2d(out_channels, out_channels, kernel_size=3, stride=1, padding=1)
+        if not dropout_after_conv:
```

Comment:
We won't add training functionality now, so let's maybe just remove all the dropouts for now :-)